Uep ov2020 2

mayavijx opened the pull request:
Updated the dockerfile with OV version 2020.2.

smkarlap commented on Azure-IoT-Edge/OnnxRuntime/UnifiedDocker/Dockerfile:
```diff
@@ -42,25 +46,25 @@ RUN tar -xzf l_openvino_toolkit*.tgz && \
-    cd /opt/intel/openvino/install_dependencies && ./_install_all_dependencies.sh && dpkg -i *.deb && \
-    pip install networkx==2.3 test-generator==0.1.1 defusedxml>=0.5.0
+    cd /opt/intel/openvino_2020.2.120/install_dependencies && ./_install_all_dependencies.sh && dpkg -i *.deb && \
+    pip3 install networkx==2.3 test-generator==0.1.1 defusedxml>=0.5.0
```
These are Model Optimizer dependencies and are no longer needed. This line should be removed.

smkarlap commented on Azure-IoT-Edge/OnnxRuntime/UnifiedDocker/Dockerfile:
```diff
@@ -42,25 +46,25 @@ RUN tar -xzf l_openvino_toolkit*.tgz && \
-ARG ONNXRUNTIME_REPO=https://github.com/microsoft/onnxruntime
-ARG ONNXRUNTIME_BRANCH=master
+ARG ONNXRUNTIME_REPO=https://github.com/intel/onnxruntime
+ARG ONNXRUNTIME_BRANCH=openvino-ep-v2
```
Should point to 'master' in github.com/microsoft/onnxruntime as UEP is merged into it now.

smkarlap commented on Azure-IoT-Edge/OnnxRuntime/UnifiedDocker/Dockerfile:
```diff
@@ -70,12 +74,12 @@ RUN wget https://github.com/Kitware/CMake/releases/download/v3.13.2/cmake-3.13.2
+    cd /onnxruntime/cmake/external/onnx && python3 setup.py install
 
-RUN cd /onnxruntime && ./build.sh --config RelWithDebInfo --update --build --parallel --use_openvino $DEVICE --build_wheel
+RUN cd /onnxruntime && ./build.sh --config RelWithDebInfo --update --build --parallel --use_openvino $DEVICE --build_wheel --use_full_protobuf
```
Build with config option 'Release' instead of 'RelWithDebInfo' to reduce container size.

smkarlap commented:
Hi Maya, Is this ready for review again?

mayavijx commented:
Hi Manohar,
It is ready for review now.
The changes made are:
1. Made config option for onnxruntime to Release.
2. Removed model optimizer dependencies.
3. Aggregate all RUN commands into one.
4. Removed the cmake 3.14 which is installed by onnxruntime/dockerfiles/scripts/install_common_deps.sh
5. Removed onnxruntime folder.

Now the image size is 2.07GB.
